test/modern_bpf/test_suites/syscall_enter_suite/mprotect_e.cpp

```diff
@@ -0,0 +1,50 @@
+#include "../../event_class/event_class.h"
+
+#ifdef __NR_mprotect
+
+#include <sys/mman.h>
+
+TEST(SyscallEnter, mprotectE)
+{
+	auto evt_test = get_syscall_event_test(__NR_mprotect, ENTER_EVENT);
+
+	evt_test->enable_capture();
+
+	/*=============================== TRIGGER SYSCALL  ===========================*/
+
+	void *mock_addr = (void *)1;
+	size_t mock_size = 4096;
+	int mock_prot = PROT_READ | PROT_EXEC;
+	assert_syscall_state(SYSCALL_FAILURE, "mprotect", syscall(__NR_mprotect, mock_addr, mock_size, mock_prot));
+
+	/*=============================== TRIGGER SYSCALL ===========================*/
+
+	evt_test->disable_capture();
+
+	evt_test->assert_event_presence();
+
+	if(HasFatalFailure())
+	{
+		return;
+	}
+
+	evt_test->parse_event();
+
+	evt_test->assert_header();
+
+	/*=============================== ASSERT PARAMETERS  ===========================*/
+
+	/* Parameter 1: addr (type: PT_UINT64) */
+	evt_test->assert_only_param_len(1, sizeof(uint64_t));
```

Comment:
I was (mistakenly) under the assumption that asserting a numeric value for an address would be useless in these tests. Since we are the ones constructing the value of the address, though, we can indeed make it deterministic! I've updated this to use ->assert_numeric_param() instead.